File: ProjEuler4.pl
#!/usr/bin/perl
# https://projecteuler.net/problem=4

# largest palindromic number that's a product of three digit numbers

use warnings;
use strict;

my($i,$j,$p,$maxNum)=(0,0,0,0);

for $i (1..999){
    for $j (($i+1)..999){
        $p = $i*$j;
        if ( ($p > $maxNum) && ("$p" eq reverse "$p") ){
            $maxNum = $p;
        }
    }
}

print "The answer is: $maxNum\n";

exit;

File: Python2/ProjectEuler/ProjEuler4.py
#!/usr/bin/python
# https://projecteuler.net/problem=4

# largest palindromic number that's a product of three digit numbers
# This is the naive approach 

isPal = lambda s: s == s[::-1]

maxNum = 1
for i in range(100, 1000):
    for j in range(i, 1000):
        s = str(i*j)
        if int(s) > maxNum and isPal(s):
            maxNum = int(s)

print "The answer is: %d " % maxNum